statsmodels/tsa/regimeswitching/kim_filter.py

```diff
@@ -0,0 +1,480 @@
+import numpy as np
+from scipy.stats import multivariate_normal
```

Comment:
Scipy didn't have multivariate normal until 0.14.0, and the earliest version supported in Statsmodels is 0.11.x. Since the ss-cykfs code requires 0.18.0 this isn't a problem now, but I suppose that we may want to backport the alternate filter timing eventually, so we might want to at least try to maintain compatibility. How hard would it be to use a try: except ImportError block, and implement the PDF by hand if it's not available in Scipy?